lib/membrane_rtc_engine/endpoints/webrtc/rtp_connection_allocator.ex

```diff
@@ -515,17 +550,55 @@ defmodule Membrane.RTC.Engine.Endpoint.WebRTC.RTPConnectionAllocator do
     old_probing_target = probing_target(old_state)
     target_updated? = new_probing_target != old_probing_target
 
-    if target_updated? do
-      Logger.debug("New probing target: #{new_probing_target}")
-
-      %{
+    state =
+      if target_updated? do
+        Logger.debug("New probing target: #{new_probing_target}")
+
+        %{
+          new_state
+          | prev_probing_epochs_overflow: expected_bits(old_state) - new_state.bits_sent,
+            bits_sent: 0,
+            probing_epoch_start: get_timestamp()
+        }
+      else
         new_state
-        | prev_probing_epochs_overflow: expected_bits(old_state) - new_state.bits_sent,
-          bits_sent: 0,
-          probing_epoch_start: get_timestamp()
-      }
-    else
-      new_state
+      end
+
+    cond do
+      new_probing_target == 0 and not is_nil(state.probing_timer) ->
+        :timer.cancel(state.probing_timer)
+        %{state | probing_timer: nil}
+
+      new_probing_target != 0 and old_probing_target == 0 ->
+        ensure_probing_timer_started(%{state | prev_probing_epochs_overflow: 0})
+
+      true ->
+        state
     end
   end
+
+  defp calculate_estimated_sender_rate(state) do
+    now = get_timestamp()
+
+    receivers =
+      Map.new(state.track_receivers, fn {key, value} ->
+        duration_in_s = Time.as_seconds(now - value.last_estimation_ts)
+        sender_rate = value.bits_since_last_estimation |> Ratio.div(duration_in_s) |> Ratio.ceil()
```

Comment:
Do we really need such a precision? I would favor using floats and I belive that's how most people calculate avg over time. I really don't like Ratio lib as it's error prone

Reply:
The only reason that I'm using it is that `Membrane.Time` returns Ratio. But yeah, we don't really need to be that precise otherwise. Just a regular 64 bit floating point precision should be more than enough